script/prerender/fileUtils.ts

```diff
@@ -42,3 +47,34 @@ export const createDiskCache = <K extends string, V>(prefix: string): Cache<K, V
     },
   };
 };
+
+let s3Client: S3Client | undefined;
+
+async function writeS3File(contentType: string, filepath: string, contents: string) {
+  let basePath = assertDefined(process.env.PUBLIC_URL, 'PUBLIC_URL environment variable not set');
+  if (basePath[0] === '/') { basePath = basePath.slice(1); }
+  const key = `${basePath}${filepath}`;
```

Comment:
the presence of this logic makes the function more than just writing files to s3, there aren't a lot of rexy files outside of the release directories but there are some, it would be nice to separate this logic into like... `writeS3File` and `prefixReleasePath` or something

Reply:
The function has been split into these 2.